Directory.Build.props

```diff
@@ -1,28 +1,41 @@
 <Project>
+  <!-- this allows us to depend on explicit package versions or version ranges while also using local project references -->
+  <!-- justification being versioning different packages differently depending on what changes are made, in line with our versioning rules -->
+  <Target Name="UseExplicitPackageVersions" BeforeTargets="GenerateNuspec">
+    <ItemGroup>
+      <_ProjectReferenceWithExplicitPackageVersion Include="@(ProjectReference->'%(FullPath)')" Condition="'%(ProjectReference.PackageVersion)' != ''" />
+      <_ProjectReferenceWithExactPackageVersion    Include="@(ProjectReference->'%(FullPath)')" Condition="'%(ProjectReference.ExactVersion)' == 'true'" />
+      <_ProjectReferenceWithReassignedVersion      Include="@(_ProjectReferencesWithVersions)"  Condition="'%(Identity)' != '' And '@(_ProjectReferencesWithVersions)' == '@(_ProjectReferenceWithExplicitPackageVersion)'" ProjectVersion="@(_ProjectReferenceWithExplicitPackageVersion->'%(PackageVersion)')" />
+      <_ProjectReferenceWithReassignedVersion      Include="@(_ProjectReferencesWithVersions)"  Condition="'%(Identity)' != '' And '@(_ProjectReferencesWithVersions)' == '@(_ProjectReferenceWithExactPackageVersion)'"    ProjectVersion="@(_ProjectReferencesWithVersions->'%(ProjectVersion)')" />
+      <_ProjectReferencesWithVersions              Include="@(_ProjectReferenceWithReassignedVersion)" Remove="@(_ProjectReferenceWithReassignedVersion)" />
```

Comment:
these lines are way too long - they were wrapped for a reason

Reply:
Get an editor/IDE that wraps on the screen instead of the text >:(